MetaMask/metamask-extension
# feat: added AccountWatcher as preinstalled snap and added to menu list

## **Description**
This PR adds a new button in the Add Account Menu to direct users to the Account Watcher Homepage. The new feature is hidden behind an Experimental Feature toggle that is off by default for a stealthy launch. 

Key changes include:
1. Added a new menu item in the Add Account Menu for watching Ethereum accounts.
2. Implemented an Experimental Feature toggle for "Watch Ethereum Accounts (Beta)".
3. Added metric tracking for the toggle state.
4. Ensured the toggle is disabled by default.

Note: flow and code changes based on [support creation of Bitcoin testnet accounts](https://github.com/MetaMask/metamask-extension/pull/25772)

[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/26402?quickstart=1)

## **Related issues**
Fixes: https://github.com/MetaMask/accounts-planning/issues/468#issue-2331781414

## **Manual testing steps**
1. Go to Settings > Experimental
2. Locate the "Watch Ethereum Accounts (Beta)" toggle
3. Enable the toggle
4. Navigate to the Add Account Menu
6. Verify that the new "Watch Ethereum account (Beta)" option is visible
7. Click on the new option and confirm it redirects to the Account Watcher Homepage
8. Watch an Ethereum account or ENS (mainnet)
9. Navigate back to Account Watcher Homepage and press the back arrow in the modal header. It _should_ redirect you to the default wallet page. 
10. Disable the toggle in Settings > Experimental
11. Verify that the "Watch Ethereum Account" option is no longer visible in the Add Account Menu

## **Screenshots/Recordings**

<img width="360" alt="Screenshot 2024-08-13 at 5 30 53 PM" src="https://github.com/user-attachments/assets/3422ce2b-b16a-42df-b4de-c0ff347f6e71">

<img width="354" alt="Screenshot 2024-08-14 at 10 07 13 AM" src="https://github.com/user-attachments/assets/c88250a4-c676-4d3b-80a3-09ad549ce9f2">

https://github.com/user-attachments/assets/6ea131db-f733-4258-83e7-7569bcffd062


## **Pre-merge author checklist**

## **Pre-merge reviewer checklist**
